src/hippopt/robot_planning/utilities/smooth_terrain.py

```diff
@@ -0,0 +1,475 @@
+        options: dict = None,
+        name: str = None,
+    ) -> TSmoothTerrain:
+        name = name if isinstance(name, str) else f"cylinder_{radius:.2f}x{height:.2f}"
+        options = {} if options is None else options
+
+        output = cls(
+            point_position_name=point_position_name,
+            options=options,
+            name=name,
+        )
+
+        point_position_xy = cs.MX.sym(output.get_point_position_name() + "_xy", 2)
+
+        top_expression = cls._top_expression_from_normal(
+            height, point_position_xy, top_normal_direction
+        )
+
+        shape_function = cs.Function(
+            "cylinder_shape",
+            [point_position_xy],
+            [
+                (1 / radius * point_position_xy[0]) ** 2
+                + (1 / radius * point_position_xy[1]) ** 2
+            ],
+            [point_position_xy.name()],
+            ["g"],
+            options,
+        )
+        top_surface_function = cs.Function(
+            "cylinder_top_surface",
+            [point_position_xy],
+            [top_expression],
+            [point_position_xy.name()],
+            ["pi"],
+            options,
+        )
+        output.set_terrain(
+            shape_function=shape_function,
+            top_surface_function=top_surface_function,
+            sharpness=side_sharpness,
+            offset=position,
+        )
+        return output
+
+    @classmethod
+    def plane(
+        cls,
+        normal_direction: np.ndarray = None,
+        position: np.ndarray = None,
+        point_position_name: str = None,
+        options: dict = None,
+        name: str = None,
+    ) -> TSmoothTerrain:
+        name = (
+            name
+            if isinstance(name, str)
+            else f"plane_{normal_direction[0]:.2f}x{normal_direction[1]:.2f}"
+            f"x{normal_direction[2]:.2f}"
+        )
+        options = {} if options is None else options
+
+        output = cls(
+            point_position_name=point_position_name,
+            options=options,
+            name=name,
+        )
+
+        point_position_xy = cs.MX.sym(output.get_point_position_name() + "_xy", 2)
+
+        top_expression = cls._top_expression_from_normal(
+            0, point_position_xy, normal_direction
+        )
+
+        shape_function = cs.Function(
+            "plane_shape",
+            [point_position_xy],
+            [0],
+            [point_position_xy.name()],
+            ["g"],
+            options,
+        )
+        top_surface_function = cs.Function(
+            "plane_top_surface",
+            [point_position_xy],
+            [top_expression],
+            [point_position_xy.name()],
+            ["pi"],
+            options,
+        )
+        output.set_terrain(
+            shape_function=shape_function,
+            top_surface_function=top_surface_function,
+            sharpness=1,
+            offset=position,
+        )
+        return output
+
+
+if __name__ == "__main__":
```

Comment:
leftover? Considering the importance, I would probably suggest to create a `hippot.[...].terrain` package somewhere, where you can also add a `__main__.py` with a cmdline utility, if needed.

Reply:
It was meant to be a test utility for debugging purposes. It is not a test, since you need to visually check if things look good, but at the same time, it is not something I am expecting a user to run.